### ISSUE TITLE
JavaScript: Clarify the definition of a factorial

### ISSUE THREAD

**Issue by Maddily** [TheOdinProject/curriculum#28902]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

It seems to me that the factorial definition under the **O(N!) - Factorial complexity** section in the **Time Complexity** lesson could be more precise.

The current definition is:
> A factorial is the product of the sequence of n integers. The factorial of 4(4!) is 4 * 3 * 2 * 1.

While it's clear from the "The factorial of 4(4!) is 4 * 3 * 2 * 1" part what a factorial is, the phrase "A factorial is the product of the sequence of n integers" may cause confusion. It makes it sound as though the product of any sequence of `n` integers is a factorial.
I think the definition should be clarified to make it clear what exactly a factorial of a number is.

I suggest rewording the definition to clarify that the factorial of a number `n` is the product of all integers from 1 to `n`. This would avoid any potential confusion and provide a more precise explanation.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-time-complexity

### (Optional) Discord Name

Maddie

**Comment by JoshDevHub** [comment]
Sure I'd accept a PR for this. Be sure to also make this change in the Ruby course's lesson on this topic.

I'll assign you.
